#! /bin/sh -e

# DP: Fix ctypes libffi configure on mips.

dir=
if [ $# -eq 3 -a "$2" = '-d' ]; then
    pdir="-d $3"
    dir="$3/"
elif [ $# -ne 1 ]; then
    echo >&2 "usage: `basename $0`: -patch|-unpatch [-d <srcdir>]"
    exit 1
fi
case "$1" in
    -patch)
        patch $pdir -f --no-backup-if-mismatch -p0 < $0
        ;;
    -unpatch)
        patch $pdir -f --no-backup-if-mismatch -R -p0 < $0
        ;;
    *)
	echo >&2 "usage: `basename $0`: -patch|-unpatch [-d <srcdir>]"
        exit 1
esac
exit 0

--- Modules/_ctypes/libffi/fficonfig.py.in~	2008-06-02 20:41:30.000000000 +0200
+++ Modules/_ctypes/libffi/fficonfig.py.in	2009-03-18 13:09:00.000000000 +0100
@@ -4,7 +4,7 @@
 
 ffi_platforms = {
     'MIPS_IRIX': ['src/mips/ffi.c', 'src/mips/o32.S', 'src/mips/n32.S'],
-    'MIPS_LINUX': ['src/mips/ffi.c', 'src/mips/o32.S'],
+    'MIPS': ['src/mips/ffi.c', 'src/mips/o32.S'],
     'X86': ['src/x86/ffi.c', 'src/x86/sysv.S'],
     'X86_FREEBSD': ['src/x86/ffi.c', 'src/x86/sysv.S'],
     'X86_WIN32': ['src/x86/ffi.c', 'src/x86/win32.S'],
